File: nom-api/Nom.Orch/Models/Recipe/RecipeIngredientModel.cs
// File: Nom.Orch/Models/Recipe/RecipeIngredientModel.cs

using System.ComponentModel.DataAnnotations;

namespace Nom.Orch.Models.Recipe
{
    public class RecipeIngredientModel
    {
        [Required]
        public long IngredientId { get; set; }

        public string Name { get; set; } = string.Empty;

        [Required]
        public decimal Quantity { get; set; }

        [Required]
        public long MeasurementId { get; set; }

        public string? Measurement { get; set; }

        public string? Notes { get; set; }
    }
}

File: nom-ui/src/app/recipe/models/recipe-ingredient.model.ts
// File: nom-ui/src/app/recipe/models/recipe-ingredient.model.ts

export interface RecipeIngredientModel {
    IngredientId: number;
    quantity: number;
    measurementId: number;
    // The name is included here for display purposes on the frontend,
    // but may not be needed in the final request payload to the backend.
    name?: string;
}